p2p/transport/websocket/websocket_test.go

```diff
@@ -548,3 +549,122 @@ func TestResolveMultiaddr(t *testing.T) {
 		})
 	}
 }
+
+func startListeners(tpt *WebsocketTransport) (ma.Multiaddr, []*manet.Listener, error) {
+	laddr := ma.StringCast("/ip4/127.0.0.1/tcp/0/ws")
+	listeners := make([]*manet.Listener, 2)
+
+	l, err := tpt.maListen(laddr)
+	if err != nil {
+		fmt.Println("Failed to listen on websocket due to error ", err)
+		return nil, nil, err
+	}
+	listeners[0] = &l
+
+	port := l.Addr().(*net.TCPAddr).Port
+	fmt.Println("Port allocated for listener:", port)
+	laddr = ma.StringCast(fmt.Sprintf("/ip4/127.0.0.1/tcp/%d/ws", port))
+	l1, err := tpt.maListen(laddr)
+	if err != nil {
+		fmt.Println("Failed to listen on websocket due to error ", err)
+		return nil, nil, err
+	}
+	listeners[1] = &l1
+
+	return laddr, listeners, nil
+}
+
+func TestListenerResusePort(t *testing.T) {
+	noOfClientConns := 4
+	var opts []Option
+	opts = append(opts, EnableReuseport())
+	_, u := newUpgrader(t)
+	tpt, err := New(u, &network.NullResourceManager{}, opts...)
+	require.NoError(t, err)
+	c := make(chan int, noOfClientConns)
+
+	raddr, listeners, err := startListeners(tpt)
+	//fmt.Println("Started Listeners")
+	for i := 0; i < 2; i++ {
+		go func(index int, ln *manet.Listener, ch chan int) {
+			l := *ln
+			defer l.Close()
+			/* Looping noOfClientConns times to ensure all 4 connections are handled.
+				With SO_REUSEPORT the distribution happens based on threads that are waiting on Accept call as mentioned below.
+				We cannot gaurantee when Server go-routines would block on Accept.
+				Sometimes the client routines get scheduled first causing unequal distribution of connections.
+				Ref: https://lwn.net/Articles/542629/
+				By contrast, the SO_REUSEPORT implementation distributes connections evenly across all of the threads (or processes)
+			 	that are blocked in accept() on the same port. */
+			for j := 0; j < noOfClientConns; j++ {
+				//j := 0
+				conn, err := l.Accept()
+				if err != nil {
+					fmt.Println("Server Routine-", index, " Failed accepting connection ", j, " due to error ", err)
+					return
+				}
+				require.NoError(t, err)
+				//fmt.Println("Server Routine-", index, " accepting connection-", j)
+				defer conn.Close()
+				buf := make([]byte, 6)
+				n, err := conn.Read(buf)
+				if n != 6 {
+					t.Errorf("read %d bytes, expected 6", n)
+				}
+				require.NoError(t, err)
```

Comment:
What do you think about something like this? Same goes to some of the other cases too, e.g. `conn.Write` and others.

```suggestion
				n, err := conn.Read(buf)
				require.NoError(t, err)
				require.Equal(t, 6, n)
```